src/test/ui/associated-types/issue-65774-1.stderr

```diff
@@ -16,6 +16,8 @@ LL |     type MpuConfig: MyDisplay = T;
 ...
 LL | impl MPU for S { }
    |      ^^^ the trait `MyDisplay` is not implemented for `T`
+   |
+   = note: required because of the requirements on the impl of `MyDisplay` for `<S as MPU>::MpuConfig`
```

Comment:
Okay we probably do need something different because it's already handled by "required by this bound in `MPU`" above apparently, and better? So we end up with duplication that mentions "impl" even if none is responsible.

Reply:
I think that the current code appropriately handles these cases by adding a new `DerivedObligation` variant. I'd like to unify `BuiltingDerivedObligation`, `ImplDerivedObligation` and `DerivedObligation` and add a field to differentiate between them, but I haven't done so as it'll require changing a couple of extra places that I didn't want to muck about with earlier. I can do that in this PR if you prefer (but will add to the diff).